src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        const StateBfAdd = { ...(newStates[newStates.length - 1] || state) };
+
+        newStates.push({ ...StateBfAdd, ...action.extraData });
+        break;
+      case 'removeProperties':
+        const stateBfRemove = { ...(newStates[newStates.length - 1] || state) };
+
+        action.keysToRemove.forEach((key) => {
+          if (Object.keys(stateBfRemove).includes(key)) {
```

Comment:
You don't need to test with an `if` here, if the property doesn't exist, nothing will happen.
So `delete stateBfRemove[key]` to an unknown key will just keet `stateBfRemove` as is